MvvmHelpers/WeakEventManager.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+using static System.String;
+
+namespace MvvmHelpers
+{
+	public class WeakEventManager
```

Comment:
I recommend making this class `internal`, because otherwise devs will start using it and you'll beholden to maintaining a `WeakEventManager` despite this library focusing on MVVM. I implemented it in [AsyncAwaitBestPractices](https://github.com/brminnick/AsyncAwaitBestPractices/blob/master/Src/AsyncAwaitBestPractices/WeakEventManager/WeakEventManager.cs), and I wish I had kept it `internal` so that only `AsyncCommand` would use it. Here's why:

**Story Time**

I released [AsyncAwaitBestPracitces](https://www.nuget.org/packages/AsyncAwaitBestPractices/), with `WeakEventManager` public, and shortly after its release someone asked me for `WeakEventManager<T>` for `EventHandler<T>`- sure, no problem! 

But then someone pointed out that not every `event` is an `EventHandler` and that events can also be a `delegate` (like [`NotifyPropertyChanged`](https://docs.microsoft.com/dotnet/api/system.componentmodel.inotifypropertychanged.propertychanged?view=netframework-4.8)), so [I added support for `Delegate`](https://github.com/brminnick/AsyncAwaitBestPractices/blob/master/Src/AsyncAwaitBestPractices/WeakEventManager/WeakEventManager.cs#L128).

But THEN someone pointed out that `Action` can also be used for events, and so I added support for [`event Action`](https://github.com/brminnick/AsyncAwaitBestPractices/blob/master/Src/AsyncAwaitBestPractices/WeakEventManager/WeakEventManager.cs#L39). But this was easier said than done, because now I needed to add an overloaded method for `WeakEventManager.HandleEvent` because `Action` and `Action<T>` doesn't consume an object, so I also ended up creating a custom Exception to throw when a dev uses the incorrect overloaded method for `HandleEvent`: https://github.com/brminnick/AsyncAwaitBestPractices/blob/master/Src/AsyncAwaitBestPractices/WeakEventManager/EventManagerService.cs#L58

tl;dr - save yourself the headaches and make this `internal`